react/RatingSummary.tsx

```diff
@@ -196,9 +219,42 @@ function RatingSummary() {
           args: { settings },
         })
       })
-  }, [client, productId])
+  }, [client])
+
+  useEffect(() => {
+    if (loadingTotalReviewsByProductId || !totalReviewsByProductId) return
+    const { totalReviewsByProductId: total } = totalReviewsByProductId
+
+    dispatch({
+      type: 'SET_TOTAL',
+      args: { total },
+    })
+  }, [
+    totalReviewsByProductId,
+    loadingTotalReviewsByProductId,
+    refetchTotalReviewsByProductId,
+  ])
 
   useEffect(() => {
+    if (loadingAverageRatingByProductId || !averageRatingByProductId) return
+    const { averageRatingByProductId: average } = averageRatingByProductId
+
+    dispatch({
+      type: 'SET_AVERAGE',
+      args: { average },
+    })
+  }, [
+    averageRatingByProductId,
+    loadingAverageRatingByProductId,
+    refetchAverageRatingByProductId,
+  ])
+
+  useEffect(() => {
+    eventBus.on('reviewSaved', () => {
+      refetchTotalReviewsByProductId()
+      refetchAverageRatingByProductId()
+    })
```

Comment:
you should put it outside of useEffect or use the unmount standard which uses the return on the useEffect to remove this listener

Reply:
ok, i also forgot to remove the old call, but now it's done. thank you.